src/util/request-manager.js

```diff
@@ -287,6 +287,11 @@ export default class RequestManager {
 
         successHosts[parts.hostname] = true;
 
+        if (body && typeof body.error === 'string') {
```

Comment:
I wouldn't mind if knew what the behavior needs to be. Without extra knowledge it's impossible to change and ensure other parts will not break. If the code was created in order to reject on a 404, and I changed it into the other 404 behavior (the one that resolves with `false`) it would break.

The commit doesn't help much for an outside like me because I don't know what motivated the addition of this line. My guess is that there was an API that returned 200 with an error in the body, but we would need input from the author of this code.